namer/namer.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 The KCP Authors.
```

Comment:
It looks like this code is copied from gengo and then modified. We need to adhere to the Apache license requirements:

1. Retain the original copyright from https://github.com/kubernetes/gengo/blob/4627b89bbf1bc415e134f359f597053a5d1b6553/namer/plural_namer.go#L1-L15
2. Add a comment stating where this code originally came from (i.e. the link above)
3. Add a comment documenting the changes we have made (function signature i.e. using `string` instead of `types.Type`)
4. Include a copy of the Apache license in this repo (already done)
5. Include a copy of the source project's NOTICE file, if applicable (not applicable here)

Reply:
Made the changes and have done the documenting [here](https://github.com/kcp-dev/code-generator/pull/21/files#diff-77b14b9c00be27bd10c6644ce9fd03052b2f83789a888ab689a823958a4e4ae7R14-R19).